diesel/src/pg/expression/functions.rs

```diff
@@ -687,3 +688,28 @@ define_sql_function! {
     #[cfg(feature = "postgres_backend")]
     fn daterange(lower: Nullable<Date>, upper: Nullable<Date>, bound: RangeBoundEnum) -> Daterange;
 }
+
+define_sql_function! {
+    /// Append an element to the end of an array.
+    /// # Example
+    ///
+    /// ```rust
+    /// # include!("../../doctest_setup.rs");
+    /// #
+    /// # fn main() {
+    /// #     run_test().unwrap();
+    /// # }
+    /// #
+    /// # fn run_test() -> QueryResult<()> {
+    /// #     use diesel::dsl::array_append;
+    /// #     use diesel::sql_types::{Integer, Array};
+    /// #     let connection = &mut establish_connection();
+    /// let ints = diesel::select(array_append::<Integer, Array<_>, _, _>(vec![Some(1), Some(2)], Some(3)))
+    ///     .get_result::<Vec<Option<i32>>>(connection)?;
+    /// assert_eq!(vec![Some(1), Some(2), Some(3)], ints);
+    /// #     Ok(())
+    /// # }
+    /// ```
+    #[cfg(feature = "postgres_backend")]
```

Comment:
Shouldn't this be outside of the macro call?

Reply:
I keep the pattern from here
https://github.com/diesel-rs/diesel/blob/444bab7eb1c054fbcde123c45d9fd671ebf79973/diesel/src/pg/expression/functions.rs#L9-L13

I double-check with `cargo expand`, and the output follows. The docs are in the correct position, and the feature is in all generated code but duplicated.
```rust
# cfg inside
❯ cargo expand -F postgres pg::expression::functions::array_append
    Checking diesel v2.2.0 (/.../diesel/diesel)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 1.01s

/// Append an element to the end of an array.
/// # Example
///
/// ```rust
/// # include!("../../doctest_setup.rs");
/// #
/// # fn main() {
/// #     run_test().unwrap();
/// # }
/// #
/// # fn run_test() -> QueryResult<()> {
/// #     use diesel::dsl::array_append;
/// #     use diesel::sql_types::{Integer, Array};
/// #     let connection = &mut establish_connection();
/// let ints = diesel::select(array_append::<Integer, Array<_>, _, _>(vec![Some(1), Some(2)], Some(3)))
///     .get_result::<Vec<Option<i32>>>(connection)?;
/// assert_eq!(vec![Some(1), Some(2), Some(3)], ints);
/// #     Ok(())
/// # }
/// ```
#[cfg(feature = "postgres_backend")]
#[cfg(feature = "postgres_backend")]
#[allow(non_camel_case_types)]
pub fn array_append<
    T: SingleValue,
    Arr: ArrayOrNullableArray<Inner = Nullable<T>> + SingleValue,
    a,
    e,
>(a: a, e: e) -> array_append<T, Arr, a, e>
where
    a: ::diesel::expression::AsExpression<Arr>,
    e: ::diesel::expression::AsExpression<Nullable<T>>,
{
    array_append_utils::array_append {
        a: a.as_expression(),
        e: e.as_expression(),
        T: ::std::marker::PhantomData,
        Arr: ::std::marker::PhantomData,
    }
}
#[cfg(feature = "postgres_backend")]
#[allow(non_camel_case_types, non_snake_case)]
///The return type of [`array_append()`](super::fn_name)
pub type array_append<T, Arr, a, e> = array_append_utils::array_append<
    T,
    Arr,
    <a as ::diesel::expression::AsExpression<Arr>>::Expression,
    <e as ::diesel::expression::AsExpression<Nullable<T>>>::Expression,
>;
```

And with the `cfg` outside the macro, it removes the duplicated `cfg` in the generated code.
```rust
# cfg outside
❯ cargo expand -F postgres pg::expression::functions::array_append
    Blocking waiting for file lock on build directory
    Checking diesel v2.2.0 (/home/guiss/projetos/open_source/diesel/diesel)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 22.55s

/// Append an element to the end of an array.
/// # Example
///
/// ```rust
/// # include!("../../doctest_setup.rs");
/// #
/// # fn main() {
/// #     run_test().unwrap();
/// # }
/// #
/// # fn run_test() -> QueryResult<()> {
/// #     use diesel::dsl::array_append;
/// #     use diesel::sql_types::{Integer, Array};
/// #     let connection = &mut establish_connection();
/// let ints = diesel::select(array_append::<Integer, Array<_>, _, _>(vec![Some(1), Some(2)], Some(3)))
///     .get_result::<Vec<Option<i32>>>(connection)?;
/// assert_eq!(vec![Some(1), Some(2), Some(3)], ints);
/// #     Ok(())
/// # }
/// ```
#[cfg(feature = "postgres_backend")]
#[allow(non_camel_case_types)]
pub fn array_append<
    T: SingleValue,
    Arr: ArrayOrNullableArray<Inner = Nullable<T>> + SingleValue,
    a,
    e,
>(a: a, e: e) -> array_append<T, Arr, a, e>
where
    a: ::diesel::expression::AsExpression<Arr>,
    e: ::diesel::expression::AsExpression<Nullable<T>>,
{
    array_append_utils::array_append {
        a: a.as_expression(),
        e: e.as_expression(),
        T: ::std::marker::PhantomData,
        Arr: ::std::marker::PhantomData,
    }
}
#[cfg(feature = "postgres_backend")]
#[allow(non_camel_case_types, non_snake_case)]
///The return type of [`array_append()`](super::fn_name)
pub type array_append<T, Arr, a, e> = array_append_utils::array_append<
    T,
    Arr,
    <a as ::diesel::expression::AsExpression<Arr>>::Expression,
    <e as ::diesel::expression::AsExpression<Nullable<T>>>::Expression,
>;
```

You are right, thanks!
I can open a PR later to fix the others